nucliadb_vectors/src/data_point/ops_hnsw.rs

```diff
@@ -29,6 +29,35 @@ use rand::rngs::SmallRng;
 use super::*;
 use crate::data_point::params;
 
+#[derive(Default)]
+struct BitField(Vec<u32>);
+
+impl BitField {
+    fn set(&mut self, pos: usize) {
+        let index = pos / 32;
+        let elem = match self.0.get(index) {
+            Some(e) => *e,
+            None => {
+                let add = 1 + index - self.0.len();
+                self.0.extend((0..add).map(|_| 0));
+                0
+            }
+        };
+        let bit = pos % 32;
+        let mask = 1 << bit;
+        self.0[index] = elem | mask;
+    }
+
+    fn get(&self, pos: usize) -> bool {
+        let Some(elem) = self.0.get(pos / 32) else {
+            return false;
+        };
+        let bit = pos % 32;
+        let mask = 1 << bit;
+        elem & mask > 0
+    }
+}
+
```

Comment:
Cool! Don't you think it would be better to use a crate lile [bit-set](https://crates.io/crates/bit-set) ? To not add extra complexity.

Reply:
Oh nice! Changed it. I wrote this in my last five minutes before leaving and the only thing I saw about bit-set is that not maintained banner :sweat_smile:. So it was faster to just write this. But it seems to work well enough and it's not like something like this needs much maintenance.